src/Features/Core/Portable/SimplifyInterpolation/AbstractSimplifyInterpolationDiagnosticAnalyzer.cs

```diff
@@ -63,19 +64,12 @@ private void AnalyzeInterpolation(OperationAnalysisContext context)
                 return;
             }
 
-            var locations = ImmutableArray.Create(interpolation.Syntax.GetLocation());
-
-            var severity = option.Notification.Severity;
-
-            for (var i = 0; i < unnecessaryLocations.Length; i++)
-            {
-                context.ReportDiagnostic(DiagnosticHelper.Create(
-                    i == 0 ? UnnecessaryWithSuggestionDescriptor : UnnecessaryWithoutSuggestionDescriptor,
-                    unnecessaryLocations[i],
-                    severity,
-                    additionalLocations: locations,
-                    properties: null));
-            }
+            context.ReportDiagnostic(DiagnosticHelper.CreateWithUnnecessaryLocations(
+                Descriptor,
+                unnecessaryLocations.First(),
+                option.Notification.Severity,
+                additionalLocations: ImmutableArray.Create(interpolation.Syntax.GetLocation()),
+                unnecessaryAdditionalLocations: unnecessaryLocations));
```

Comment:
feels weird that the first unnecessary location is specified in two places.  is that intentional?